### ISSUE TITLE
Ngxs developmentMode activated

### ISSUE THREAD

**Issue by paulstelzer** [ngxs/store#716]
## I'm submitting a...

<pre><code>
[ ] Regression (a behavior that used to work and stopped working in a new release)
[x] Bug report  
[ ] Performance issue
[ ] Feature request
[ ] Documentation issue or request
[ ] Support request => https://github.com/ngxs/store/blob/master/CONTRIBUTING.md
[ ] Other... Please describe:
</code></pre>

## Current behavior
It's a clear bug, but if I activate the developmentMode in `NgxsModule.forRoot`, I get errors outside of Ngxs. Is this normal? What does developmentMode  do?

I am using the firebase js sdk, too, so for example this issue is related to them too. So it's more a question what exactly developmentMode is doing.

![chrome_2018-12-17_19-36-03](https://user-images.githubusercontent.com/20501666/50107439-0612e500-0233-11e9-8a98-b079e2d492dc.jpg)

## Environment

<pre><code>
Libs:
- @angular/core version: 7.1.3
- @ngxs/store version: 3.3.3

Browser:
- [ ] Chrome (desktop) version XX
- [ ] Chrome (Android) version XX
- [ ] Chrome (iOS) version XX
- [ ] Firefox version XX
- [ ] Safari (desktop) version XX
- [ ] Safari (iOS) version XX
- [ ] IE version XX
- [ ] Edge version XX
 
For Tooling issues:
- Node version: XX  
- Platform:  

Others:

</code></pre>

**Comment by eranshmil** [comment]
developmentMode activates deep freeze of the state properties, in order to avoid mutation.
Do you store firebase SDK's objects inside your state?

Please provide a reproduction using https://stackblitz.com/fork/ngxs-simple, and maybe we could help.

**Comment by markwhitfeld** [comment]
When you enable `developmentMode` NGXS freezes any object that you put into the state to ensure that you do not mutate the state. Part of the paradigm is that you only store immutable plain old javascript objects in your State and that you honour the immutability when you do updates.
If you mutate the objects on your State then the change detection and selectors will not function correctly.
`developmentMode` is designed to identify these types of issues where the fundamental paradigm is not being followed.

**Comment by paulstelzer** [comment]
@eranshmil Your assumption was right. I saved the firebase.User inside the store. I removed this and all errors are gone away.

**Comment by markwhitfeld** [comment]
Yeah, we will hopefully add some other useful `devlopmentMode` detections in future that would warn you if you put objects that derive from classes in your store.
